src/main/kotlin/sudoklin/data/Sudoku.kt

```diff
@@ -131,6 +131,10 @@ class NewShinySudoku() {
         }
     }
 
+    fun addSolvedNumber(rowIndex: Int, columnIndex: Int, number: Int) {
+        rows[rowIndex]!!.cells[columnIndex].candidates = IntArray(1, { number })
```

Comment:
re-assinging this smells a bit. Do we have a better way for this?

Reply:
as just discussed, maybe can be a List or Set. which then would be emptied in this case, then filled with the single number.